### PR TITLE
#17: added tests with different virtual schemas

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java [hunk 1]
```diff
@@ -44,27 +49,72 @@ static void beforeAll() throws SQLException, BucketAccessException, InterruptedE
...
+    private static void createAdapterScript() throws SQLException, InterruptedException {
         statement.execute("CREATE OR REPLACE JAVA ADAPTER SCRIPT RLS_SCHEMA.ADAPTER_SCRIPT_EXASOL_RLS AS " //
                 + "%scriptclass com.exasol.adapter.RequestDispatcher;\n" //
                 + "%jar /buckets/bfsdefault/default/" + ROW_LEVEL_SECURITY_JAR_NAME_AND_VERSION + ";\n" //
                 + "/");
         TimeUnit.SECONDS.sleep(20); // FIXME: need to be fixed in the container
-        statement.execute("CREATE VIRTUAL SCHEMA virtual_schema_rls USING RLS_SCHEMA.ADAPTER_SCRIPT_EXASOL_RLS " //
+    }
+
+    private static void createVirtualSchemaJdbc() throws SQLException {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java]: Reduce redundancy by adding a parameter that accepts a key-value map to the create script.

File: src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java [hunk 2]
```diff
@@ -176,20 +226,25 @@ private static void createUser(final String userName) throws SQLException {
         statement.execute("GRANT ALL PRIVILEGES TO " + userName + "");
     }
 
-    @Test
-    void testSelectFromExaRlsUsersThrowsException() {
-        assertThrows(SQLException.class, () -> statement.execute("SELECT * FROM EXA_RLS_USERS"));
+    @ParameterizedTest
+    @ValueSource(strings = { VIRTUAL_SCHEMA_RLS_JDBC_NAME, VIRTUAL_SCHEMA_RLS_JDBC_LOCAL_NAME,
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java]: Use a method source instead to reduce redundancy.

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java]: done

File: src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java]: done

File: src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java [hunk 3]
```diff
@@ -171,25 +194,35 @@ private static void createRolesAndTenantsTestTable() throws SQLException {
...
+        for (String userName : userNames) {
+            statement.execute("CREATE USER " + userName + " IDENTIFIED BY \"" + userName + "\"");
+            statement.execute("GRANT ALL PRIVILEGES TO " + userName + "");
+        }
     }
 
-    @Test
-    void testSelectFromExaRlsUsersThrowsException() {
-        assertThrows(SQLException.class, () -> statement.execute("SELECT * FROM EXA_RLS_USERS"));
+    private static Stream<String> withMethodSource() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java]: Please rename to `getVirtualSchemaVariants`.

**Review by redcatbear — APPROVED** [review]
Approved with one finding to be fixed.

File: src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityDialectIT.java]: fixed

**Pull request merged**